src/App.ts

```diff
@@ -411,14 +411,15 @@ export default class App {
     // Set body and payload (this value will eventually conform to AnyMiddlewareArgs)
     // NOTE: the following doesn't work because... distributive?
     // const listenerArgs: Partial<AnyMiddlewareArgs> = {
-    const listenerArgs: Pick<AnyMiddlewareArgs, 'body' | 'payload'> & {
+    const listenerArgs: Pick<AnyMiddlewareArgs, 'logger' | 'body' | 'payload'> & {
```

Comment:
everything looks good to me. I'm not sure if there are any unintended ramifications adding logger here and to 422. I'd suggest @aoberoi take a quick look.